server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/AliasRoutes.java

```diff
@@ -136,4 +164,60 @@ private void ensureUserDoesNotExist(MailAddress mailAddress) throws UsersReposit
+                MAILADDRESS_ASCII_DISCLAIMER),
+        @ApiImplicitParam(required = true, dataType = "string", name = ALIAS_SOURCE_ADDRESS, paramType = "path",
+            value = "Source mail address of the alias to remove.\n" +
+                MAILADDRESS_ASCII_DISCLAIMER)
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "OK"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400,
+            message = ALIAS_DESTINATION_ADDRESS + " or forward structure format is not valid"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+            message = "Internal server error - Something went bad on the server side.")
+    })
+    public HaltException removeFromAliasSources(Request request, Response response) throws RecipientRewriteTableException {
+        MailAddress destinationAddress = MailAddressParser.parseMailAddress(request.params(ALIAS_DESTINATION_ADDRESS), ADDRESS_TYPE);
+        MailAddress aliasToBeRemoved = MailAddressParser.parseMailAddress(request.params(ALIAS_SOURCE_ADDRESS), ADDRESS_TYPE);
+        MappingSource source = MappingSource.fromUser(User.fromMailAddress(destinationAddress));
+        recipientRewriteTable.removeAliasMapping(source, aliasToBeRemoved.asString());
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    @GET
+    @Path(ROOT_PATH + "/{" + ALIAS_DESTINATION_ADDRESS + "}")
+    @ApiOperation(value = "listing alias sources of an address")
+    @ApiImplicitParams({
+        @ApiImplicitParam(required = true, dataType = "string", name = ALIAS_DESTINATION_ADDRESS, paramType = "path")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK", response = List.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "The destination is not an address"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+            message = "Internal server error - Something went bad on the server side.")
+    })
+    public ImmutableSet<AliasSourcesResponse> listAliasesOfAddress(Request request, Response response) throws RecipientRewriteTableException {
+        MailAddress baseAddress = MailAddressParser.parseMailAddress(request.params(ALIAS_DESTINATION_ADDRESS), ADDRESS_TYPE);
+        Mappings mappings = recipientRewriteTable.getStoredMappings(MappingSource.fromMailAddress(baseAddress))
+            .select(Mapping.Type.Alias);
+
+        if (mappings == null || mappings.isEmpty()) {
+            return ImmutableSortedSet.of();
+        }
+
+        return mappings.asStream()
+            .map(mapping -> mapping.asMailAddress()
+                .orElseThrow(() -> new IllegalStateException(String.format("Can not compute address for mapping %s", mapping.asString()))))
+            .map(MailAddress::asString)
+            .sorted()
+            .map(AliasSourcesResponse::new)
+            .collect(Guavate.toImmutableSet());
```

Comment:
`Guavate` -> `ImmutableSet`

Reply:
Nope cf it"s experimental & @mbaechler did ask to keep using guavate